src/sensors/elec.rs

```diff
@@ -14,21 +26,142 @@ pub fn filter_pde(pde: f32) -> bool {
     random::<f32>() < pde
 }
 
+pub fn gaussian_sampler(mu: f32, sigma: f32) -> Box<dyn Fn() -> f32> {
+    let dist = Normal::new(mu, sigma).unwrap();
+    Box::new(move || dist.sample(&mut rand::thread_rng()))
+}
+
+// Could this be a FROM? How to use the Group here to avoid too many collects?
+fn qt_min_time(event_id: u32, sensor_id: u32, hits: Vec<MCSensorHit>) -> MCQT {
```

Comment:
I have a bit of a doubt related to your critique of the group version. Maybe I'm confused because the code had already changed before I touched it.

The algorithm requires that we have all information for a given event (coincidence in real terms, I suppose but in MC it's an event). Given the way the tables are written in our MC output we have to read all the information for hits into memory to be able to group them (at least how the readers are written at the moment). How is this different to the way we have things int he Julia code? Other than we put the information into a vector of structs here which we then group using `itertools` it seems the same to me. Or you think that there's an efficient way to change the readers so they yield iterators to the information up to the change of event?

Reply:
All I"m saying is that you should be aware that group-by works differently in data frames from how it works in iterators.

Specifically, given the data

| index | x |
|-------|----|
| 1 | a |
| 2 | b |
| 3 | a |

group-by(x) will give you 2 groups in the case of data frames, and three groups in the case of iterators:

+ data frames: `{1,3}` and `{2}`
+ iterators: `{1}`, `{2}` and `{3}`


Somewhat related to this, are two other potential problems I'm aware of:

1. Geant4 spits out hits out of time order. This leads to bugs if one assumes that they are time-ordered. But I think that I take care of that before writing the data out from G4, so it shouldn't be an issue here.
2. Some of the algorithms we have (or had in the past) rely on event numbers being monotonically non-decreasing. This requires care when we combine events from multiple MC output files (as we do in `makelor`) especially if paralellization is thrown into the mix.

In some cases, the bugs cased by this might be subtle: the code may continue to produce results, but the quality might be degraded.